PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/CreateFunctionAction.java

```diff
@@ -85,41 +72,61 @@ public CreateFunctionAction() {
         String hintPackageName = pkg == null ? "" : pkg.getQualifiedName();
 
         CreateFunctionForm form = new CreateFunctionForm(project, hintPackageName);
         List<PsiElement> psiElements = new ArrayList<>();
         if (form.showAndGet()) {
-            Map<String, String> parameters = form.getTemplateParameters();
-            String triggerType = form.getTriggerType();
-            String packageName = parameters.get("packageName");
-            String className = parameters.get("className");
-
-            PsiDirectory directory = com.intellij.psi.util.ClassUtil.sourceRoot(psiDirectory);
-            String newName = packageName.replace('.', '/');
             final FunctionTemplate bindingTemplate;
             try {
+                Map<String, String> parameters = form.getTemplateParameters();
+                final String connectionName = parameters.get("connection");
+                String triggerType = form.getTriggerType();
+                String packageName = parameters.get("packageName");
+                String className = parameters.compute("className", (k, v) -> AzureFunctionsUtils.normalizeClassName(v));
+                PsiDirectory directory = ClassUtil.sourceRoot(psiDirectory);
+                String newName = packageName.replace('.', '/');
                 bindingTemplate = form.getFunctionTemplate(triggerType);
-                final String functionClassContent = AzureFunctionsUtils.substituteParametersInTemplate(bindingTemplate, parameters);
-                Application application = ApplicationManager.getApplication();
-                application.runWriteAction(() -> {
-                    CreateFileAction.MkDirs mkDirs = ApplicationManager.getApplication().runWriteAction((Computable<CreateFileAction.MkDirs>) () -> new CreateFileAction.MkDirs(newName + '/' + className, directory));
-                    PsiFileFactory factory = PsiFileFactory.getInstance(project);
-                    CommandProcessor.getInstance().executeCommand(project, () -> {
-                        PsiFile psiFile = factory.createFileFromText(className
-                                + ".java", JavaFileType.INSTANCE, functionClassContent);
-
-                        psiElements.add(mkDirs.directory.add(psiFile));
-                    }, null, null);
-
-                    if (StringUtils.equalsIgnoreCase(triggerType, CreateFunctionForm.EVENT_HUB_TRIGGER)) {
-                        String connectionString = getEventHubNamespaceConnectionString(form.getEventHubNamespace());
-                        saveLocalSetting(project, parameters.get("connection"), connectionString);
-                    }
-                });
 
+                if (StringUtils.equalsIgnoreCase(triggerType, CreateFunctionForm.EVENT_HUB_TRIGGER)) {
+                    if (StringUtils.isBlank(connectionName)) {
+                        throw new AzureExecutionException("Required property <connection> is missing");
```

Comment:
I think we should change the wording like `Failed to connect event hub` or something else as it's ide toolkit who get the connection string from Azure, there is no way for users to set the connection. If we add the support for users to set the connection string, we should move the validation `CreateFunctionForm`

Reply:
That will be laid inside CreateFunctionForm in later PR